### DISCUSSION TITLE
[IntelliSense] Show classes IntelliSense by pressing custom shortcut key combination

### DISCUSSION THREAD

**Discussion by inf1nite-lo0p** [tailwindlabs/tailwindcss#6960] [Ideas]
This feature request might also solve the following issues:
tailwindlabs/tailwindcss-intellisense#350
tailwindlabs/tailwindcss-intellisense#393 

First of all, let me say thanks for this amazing extension, in addition, I'm not even sure if it's possible to do this feature with the current vscode extensions API.

It would be awesome if we could get IntelliSense anywhere we need by pressing a custom key combination such as `ctrl`+`shift`+`space`. The shortcut key combination should be disabled by default and can be configured through extension settings.

As a React developer most of the time, I'm using tailwind classes in `switch` or `if-else` statements, I might even need to pass them as other props such as `wrapperClassName` or using function arguments.

I believe having the ability to get tailwindcss's class names by pressing a custom key combination can solve all of the above issues.

**Comment by caitlynrw** [comment]
I'm pretty sure VSCode already has a feature that allows you to bring up the intellisense menu whenever you want.

**Comment by crswll** [comment]
This is a feature I'd like too.

One thing Brad showed me which has eased the pain a bit is:

```
"tailwindCSS.experimental.classRegex": [
  "const className\\s*=\\s*'([^']*)'"
]
```
and this will trigger it to show when `const className = "you are in here"`.

I'm kind of tempted to just have a dummy `const tw = s => s` and do:

```
"tailwindCSS.experimental.classRegex": [
  "tw\\('([^)]*)'\\)"
]
```

and just author my classes like: `const whatever = tw('auto complete here')`. Not elegant by any means but low effort.

**Comment by crswll** [comment]
You know I should have looked at that second issue before I posted my comment.

**Comment by sylann** [comment]
I think this issue was solved when tailwindlabs/tailwindcss-intellisense#350 was.
It should probably be closed.

**Comment by apolzon** [comment]
> It would be awesome if we could get IntelliSense anywhere we need by pressing a custom key combination such as `ctrl`+`shift`+`space`. The shortcut key combination should be disabled by default and can be configured through extension settings.

Sorry to ping an old issue, but it would be great if this request to offer a keybinding for the generated CSS output was added. Unless I'm missing something, the hover popover is different than asking vscode to re-open the suggestion box. Furthermore, if you try to accomplish this by re-opening the suggestion box, your cursor has to be at the end of the classname or it will attempt to autocomplete from the characters before your cursor (which is the correct behavior) -- all that to say, I don't think re-opening the suggestion box is the same as the hover feature and re-opening suggestions is not really a great workaround.

Edit: Sorry, I'm an idiot and realized there most definitely IS a way to display the hover popup via keybind in VSCode. I just misunderstood the difference between it and the autosuggestions. Apologies!
(And for anyone stumbling on this, the default keybind is cmd+k cmd+i -- it can be rebound under the key `editor.action.showHover`)
